reactive/calico.py

```diff
@@ -219,9 +220,20 @@ def blocked_without_etcd():
 def install_etcd_credentials():
     etcd = endpoint_from_flag('etcd.available')
     etcd.save_client_credentials(ETCD_KEY_PATH, ETCD_CERT_PATH, ETCD_CA_PATH)
+    # register initial etcd data so that we can detect changes
+    data_changed('calico.etcd.data', (etcd.get_connection_string(),
+                                      etcd.get_client_credentials()))
     set_state('calico.etcd-credentials.installed')
 
 
+@when('etcd.tls.available', 'calico.service.installed')
+def check_etcd_changes():
+    etcd = endpoint_from_flag('etcd.available')
+    if data_changed('calico.etcd.data', (etcd.get_connection_string(),
+                                         etcd.get_client_credentials())):
+        remove_state('calico.service.installed')
```

Comment:
This will lead to a restart of Calico, but it won't lead to the etcd certs being rewritten to disk (which happens [here](https://github.com/charmed-kubernetes/layer-calico/blob/7fdf33fca54bc414610c4ae1f2f420d8c43e1faa/reactive/calico.py#L222)).

Reply:
This should be addressed with the latest commit.